client/state/ui/reducer.js

```diff
@@ -29,6 +29,26 @@ export function selectedSiteId( state = null, action ) {
 	return state;
 }
 
+/**
+ * Tracks the currently selected site ID.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function setRecentlySelectedSiteId( state = [], action ) {
+	switch ( action.type ) {
+		case SELECTED_SITE_SET:
+			state = [ action.siteId, ...state ];
+			if ( state.length > 4 ) {
+				state.pop();
```

Comment:
might be safer to do `state.length = 4` to ensure the truncation is doing what you expect.

could also slice that above, taking the first 3 elements, and unshift the new siteId.